yarl/_url.py

```diff
@@ -1153,7 +1158,52 @@ def join(self, url):
         # See docs for urllib.parse.urljoin
         if not isinstance(url, URL):
             raise TypeError("url should be URL")
-        return URL(urljoin(str(self), str(url)), encoded=True)
+        other: URL = url
+        scheme = other.scheme or self.scheme
+        parts = {
+            k: getattr(self, k) or ""
+            for k in ["authority", "path", "query_string", "fragment"]
+        }
+        parts["scheme"] = scheme
+
+        if scheme != self.scheme or scheme not in uses_relative:
+            return URL(str(other))
+
+        if scheme in uses_authority:
```

Comment:
It looks like there is a branch miss on coverage for this line.

Reply:
> 774, 1172->1182, 1268-1269

I tried, setting a breakpoint at 1173 or 1170 works for me?

Reply:
If codecov is correct, then we're also missing lines 1164-1165 an `if not segments` on line 773 is never True anymore.

Reply:
I don't think it's possible for 1164-1165 to not run, so I that's a bug in the coverage...

Reply:
```python
        parts = {
            k: getattr(self, k) or ""
            for k in ["authority", "path", "query_string", "fragment"]
        }
```
guess coverage does not like the `or ""`